modules/host/contractmanager/persist_test.go

```diff
@@ -26,6 +26,365 @@ func (*dependencyNoRecheck) Disrupt(s string) bool {
 	return false
 }
 
+func Test_savedStorageFolder_Equals(t *testing.T) {
+	tests := []struct {
+		name     string
+		folder   savedStorageFolder
+		expected savedStorageFolder
+		want     bool
+	}{
+		{
+			name: "equal",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: true,
+		},
+		{
+			name: "diff path",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/02",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "diff index",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 0,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "diff usage",
+			folder: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					1,
+					2,
+					3,
+					4,
+					5,
+				},
+			},
+			expected: savedStorageFolder{
+				Index: 1,
+				Path:  "/tmp/storage/01",
+				Usage: []uint64{
+					5,
+					1,
+					2,
+					3,
+					4,
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.folder.Equals(tt.expected); got != tt.want {
+				t.Errorf("savedStorageFolder.Equals() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_savedSettings_Equals(t *testing.T) {
```

Comment:
conventionally we don't use `_` in test names; everything is squished together and capitalized if necessary. I can't think of this ever causing issues in practice, because it's very rare to have two types or methods with the same name (where one is exported).